opnode/l2/driver.go

```diff
@@ -0,0 +1,208 @@
+package l2
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/event"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/eth"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type EngineDriver struct {
+	Log log.Logger
+	// API bindings to execution engine
+	RPC DriverAPI
+
+	SyncRef SyncReference
+
+	// The current driving force, to shutdown before closing the engine.
+	driveSub ethereum.Subscription
+	// There may only be 1 driver at a time
+	driveLock sync.Mutex
```

Comment:
Shouldn't the property of one driver per engine be enforced at the point where the `EngineDriver` gets instantiated rather than allowing multiple `EngineDriver` for the same execution engine then using a mutex?

Reply:
Being able to manage the engine without actively running the driver process will be useful later, e.g.:
- Not running the default driver during state sync
- Stopping/starting the driver for integration tests to force nodes out-of-sync